docs/madr/decisions/006-mesh-traffic-permission.md

```diff
@@ -0,0 +1,503 @@
+    name: ...
+  from:
+    - targetRef:
+        kind: Mesh|MeshSubset|MeshService|MeshServiceSubset
+        name: ...
+      conf:
+        action: ALLOW|DENY
+```
+
+We don't want to mix mTLS configs (like permissive mTLS or TLS version) with MeshTrafficPermission. 
+The main reason is that mTLS settings are applied for the entire inbound, while permissions could be set for a group of clients. 
+
+### Envoy configuration
+
+Envoy has [RBAC Network Filter](https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/network_filters/rbac_filter) 
+that allows configuring permissions for the list of principals.
+
+**Principal** is an entity that can be authenticated.<br>
+**Permission** is access granted to the principal.  
+
+Each Kuma service has a generated certificate with URIs that identify service name and tags.
+For example "backend" service in "us-east" zone of version v1 can have the following list of URIs:
+* spiffe://mesh-1/backend
+* kuma://kuma.io/zone/us-east
+* kuma://version/v1
+
+If we want to grant any permissions to these instances we can create the following Network RBAC:
+
+```yaml
+action: ALLOW
+policies:
+  "backend-us-east-v1-allow-all":
+    permissions:
+      - any: true
+    principals:
+      - and_ids:
+          ids:
+            - authenticated:
+                principal_name:
+                  exact: "spiffe://mesh-1/backend"
+            - authenticated:
+                principal_name:
+                  exact: "kuma://kuma.io/zone/us-east"
+            - authenticated:
+                principal_name:
+                  exact: "kuma://version/v1"
+```
+
+In order request to be allowed, at least 1 policy should be matched. 
+
+#### Converting MeshTrafficPermission to Envoy NetworkRBAC
+
+1. Concatenate "from" arrays from all matched MeshTrafficPermissions 
+
+Input
+```yaml
+type: MeshTrafficPermission
+name: global
+spec:
+  targetRef:
+    kind: Mesh
+  from:
+    - targetRef:
+        kind: Mesh
+      conf:
+        action: DENY
+    - targetRef:
+        kind: MeshSubset
+        tags:
+          kuma.io/zone: us-east
+      conf:
+        action: ALLOW
+---
+type: MeshTrafficPermission
+name: backend
+spec:
+  targetRef:
+    kind: MeshService
+    name: backend
+  from:
+    - targetRef:
+        kind: MeshSubset
+        tags:
+          env: dev
+      conf:
+        action: DENY
+```
+
+Concatenated "from" array
+```yaml
+from:
+  - targetRef:
+      kind: Mesh
+    conf:
+      action: DENY
+  - targetRef:
+      kind: MeshSubset
+      tags:
+        kuma.io/zone: us-east
+    conf:
+      action: ALLOW
+  - targetRef:
+      kind: MeshSubset
+      tags:
+        env: dev
+    conf:
+      action: DENY
+```
+
+2. Create a full rule-based view (with negations) 
+
+Regular [rule-based view](005-policy-matching.md#rule-based-view) doesn't work well for NetworkRBAC generation,
+because rule-based view relies on the order, but NetworkRBAC doesn't. 
```

Comment:
But it can rely on order, no?
"The policies are evaluated in lexicographic order of the policy name"
https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/rbac/v3/rbac.proto#envoy-v3-api-msg-config-rbac-v3-rbac
we pick a name for the policies in NetworkRBAC filter. We can sort them then however we want.

Reply:
I see. Ok.

Reply:
I remember order being important when rules overlap. E.g.

allow: /* 
deny: /some-path/*

The `deny` in this case is useless. Have you taken into account the overlapping of rules?